### PR TITLE
roachtest: small ldr roachtest

### PR DESCRIPTION
**Pull request by stevendanna**
This is a small roachtest for LDR. It would be nice to share more infrastructure with the cluster to cluster tests, but I am hoping that happens a bit more organically as we develop the corpus of LDR tests.

Epic: none
Release note: None

### CODE REVIEW

**Comment by stevendanna** [comment]
In follow-up PRs I'll be adding table verification.

File: pkg/cmd/roachtest/tests/jobs_util.go [hunk 1]
```diff
@@ -191,6 +191,45 @@ func WaitForRunning(
 	}, maxWait)
 }
 
+type jobRecord struct {
+	status   string
+	payload  jobspb.Payload
+	progress jobspb.Progress
+}
+
+func (jr *jobRecord) GetHighWater() time.Time {
```
Comment (msbutler) [inline pkg/cmd/roachtest/tests/jobs_util.go:200]: i'm contemplating starting a dumb thread with cdc about the term "high water mark". They bind the _frontier_ of a span frontier to their persisted high water mark here:
https://github.com/msbutler/cockroach/blob/master/pkg/ccl/changefeedccl/changefeed_processors.go#L781

This confused me for several years. I feel like we should universally use the term low water mark for the lowest timestamp of a span frontier (i.e. the frontier), and the highest timestamp in the frontier as the high water mark. I think clearing this up will help everyone out, especially the new hires.

But maybe i'm missing some context.

**Review by msbutler — APPROVED** [review]
Nice! Before merging, I recommend running a c2c and a cdc roachtest to make sure your changes all place nice.

File: pkg/cmd/roachtest/tests/logical_data_replication.go [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+			name: "ldr/kv0/workload=left/ingestion=both",
+			clusterSpec: multiClusterSpec{
+				leftNodes:  3,
+				rightNodes: 3,
+				clusterOpts: []spec.Option{
+					spec.CPU(8),
+					spec.VolumeSize(100),
+				},
+			},
+			run: func(ctx context.Context, t test.Test, c cluster.Cluster, setup multiClusterSetup) {
```
[the remark this answers is not in the document]
Comment (stevendanna) [reply pkg/cmd/roachtest/tests/logical_data_replication.go:47]: 👍 Will be pushing some more tests this week where I think we'll want to abstract a bit like you've done for the c2c tests.

**Review by msbutler — APPROVED** [review]

**Comment by stevendanna** [comment]
@msbutler Thanks for the re-review :D

**Pull request merged**